### PR TITLE
Unit test for LocalStateQuery

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

Relates to #1577 / ADP-83.

### Overview

Adds a test of NetworkLayer.getTxParameters against the self-node

### CODE REVIEW

File: lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+spec = describe "getTxParameters" $ do
+    it "Correct values are queried" $ do
+        withTestNode $ \gbp sock vData -> withLogging $ \(tr, getLogs) -> do
+            let gbp' = gbp { txParameters = fakeTxParameters }
+            withNetworkLayer tr gbp' sock vData $ \nl -> do
+                let retryPolicy = constantDelay 1_000_000 <> limitRetries 10
+                recoverAll retryPolicy $ const $
+                    getTxParameters nl `shouldReturn` txParameters gbp
+            msg <- mapMaybe isMsgTxParams <$> getLogs
+            msg `shouldBe` [txParameters gbp]
```
Comment (KtorZ) [inline lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs:54]: Feels more like an integration test than a unit test though.

**Review by KtorZ — APPROVED** [review]

File: lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs (hunk 1 above)
Comment (rvl) [reply lib/byron/test/unit/Cardano/Wallet/Byron/NetworkSpec.hs:54]: The byron network layer is not set up very well for unit testing, so using the real cardano-node was the best that I could do.

**Comment by rvl** [comment]
Socket file issue on macos - fixed.

bors r+

**Pull request merged**